### ISSUE TITLE
TF2 model output NaN when inferenced is performed using CPU

### ISSUE THREAD

**Issue by talhaanwarch** [tensorflow/tensorflow#48205]
I have a TensorFlow model architecture and pre-trained weights. When I use it for prediction on GPU it works well, but on CPU it starts giving NaNs. The model weights are in hdf5 format. There is no batch normalization layer in the model. I am confused why it's not working on the CPU.  
Tensorflow version:2.30.  
Tested on Windows and Linux (same issue on both). A similar error is reported in #35307 . But in TensorFlow version 2, i am not sure how to use clear_device as mentioned in [the comment of this issue](https://github.com/tensorflow/tensorflow/issues/35307#issuecomment-568136811). The model I am using is available [here  ](https://github.com/xliucs/MTTS-CAN/blob/5db0928c7474bc2206d6132cd79b09aadc619897/code/model.py#L203)

**Comment by amahendrakar** [comment]
@talhaanwarch,
The `model.py` file you've linked has only the class and function definitions. I did not get any output on running the code. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/045a3d9ba253e11ef575bfd7ebd77a78/48205.ipynb). 

In order to reproduce the issue reported here, could you please provide a minimal code snippet and the dataset you are using. Thanks!

**Comment by talhaanwarch** [comment]
Please check this
https://colab.research.google.com/drive/16RbTjME6LcGjG7uSOSr5f8KQMXBH65WZ?usp=sharing

Run it with CPU and with GPU

**Comment by talhaanwarch** [comment]
@amahendrakar  is there any update regarding this

**Comment by amahendrakar** [comment]
@jvishnuvardhan,
I was able to reproduce the issue with TF v2.3 and TF v2.4. Whereas Colab doesn't detect GPU with TF v2.5.0rc0 and TF-nightly.

Code runs find on [GPU](https://colab.research.google.com/gist/amahendrakar/cac92aa388ec1b744ee804187ef29551/48205-gpu.ipynb), but gives NaN predictions with [CPU](https://colab.research.google.com/gist/amahendrakar/079fc1a475a8411a1ebe8aa79a883f4c/48205-cpu.ipynb). Please check the linked gist for reference. Thanks!

**Comment by saikumarchalla** [comment]
Was able to reproduce the issue in TF 2.5 stable version. Please find the gist [here](https://colab.research.google.com/gist/saikumarchalla/6f0d1aec8acb3c1938c7f0214ecf20bc/48205-cpu.ipynb).Thanks!

**Comment by ch3rub1m** [comment]
I have met exactly the same bug, all is right when using GPU but nan in CPU
